### DISCUSSION TITLE
Inventory Updates

### DISCUSSION THREAD

**Discussion by jvanderaa** [nautobot/nornir-nautobot#37] [Ideas]
For the future of the Nornir Inventory for Nautobot, I am thinking that it may be best to move to using GraphQL all together. This could greatly reduce the time spent to gather inventory data. If we do this, we may not be able to have the pynautobot object passed back, also we may need to remove the pynautobot data dictionary that is in the inventory today. 

As an alternative, we could write a second inventory that uses just GraphQL with those features and not have to remove functionality that may be beneficial for some organizations. In the spirit of an inventory is just a starting point for the data, that we don't need to have the data, this should be just fine. 

Thoughts?

**Comment by jeffkala** [comment]
Meaning a base class and then just a graphql and standard class that inherit from it?  Or am I miss interpreting?  This route might provide a way to do a slower deprecation over time?

**Reply by jvanderaa** [reply]
Interesting thought there. I like the idea.
